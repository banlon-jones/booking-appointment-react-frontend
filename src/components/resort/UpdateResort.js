import React from 'react';
import { useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';
import AddResort from './AddResort.module.css';
import setUpdateResort from '../../store/resort-details/updateResort';
import Form from './resort_components/Form';

function UpdateResort() {
  const dispatch = useDispatch();
  const { id } = useParams();
  const updateResortValue = (e) => {
    e.preventDefault();
    const resort = {
      name: e.target.name.value,
      country: e.target.country.value,
      city: e.target.city.value,
      description: e.target.description.value,
      cost: e.target.cost.value,
      image: e.target.image.value,
    };
    dispatch(setUpdateResort(resort, id));
  };

  return (
    <div className={AddResort.add_resort_container}>
      <h1>Update Resort</h1>
      <Form updateResortValue={updateResortValue} formId="update" />
    </div>
  );
}

export default UpdateResort;
