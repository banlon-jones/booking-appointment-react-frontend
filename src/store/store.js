/* eslint-disable import/no-named-as-default */
/* eslint-disable import/no-named-as-default-member */
import { configureStore } from '@reduxjs/toolkit';
import userSlice from './user/user';
import resortDetailSlice from './resort-details/resortDetail';

const store = configureStore({
  reducer: {
    user: userSlice,
    resortDetail: resortDetailSlice,
  },
});

export default store;
