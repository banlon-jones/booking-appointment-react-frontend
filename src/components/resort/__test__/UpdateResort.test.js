/* eslint-disable comma-dangle */
import React from 'react';
import { cleanup, screen, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../../store/store';
import UpdateResort from '../UpdateResort';

afterEach(cleanup);

describe('React Test for AddResort', () => {
  test('Add RssortTest', () => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <UpdateResort />
        </Provider>
      </BrowserRouter>
    );
    const ResortElement = screen.getByTestId('Update-resort');
    expect(ResortElement).toBeInTheDocument('Update Resort');
  });
});

describe('Add Resort Snapshot', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <UpdateResort />
          </Provider>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
