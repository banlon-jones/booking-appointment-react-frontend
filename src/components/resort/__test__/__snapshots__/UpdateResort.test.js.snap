// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Resort Snapshot renders correctly 1`] = `
<div
  className="add_resort_container"
  data-testid="Update-resort"
>
  <div
    className="notice_container"
    data-testid="notice"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <p
      className="notification"
      style={
        Object {
          "backgroundColor": " ",
        }
      }
    >
       
    </p>
  </div>
  <h1>
    Update Resort
  </h1>
  <form
    className="form"
    data-testid="form"
    onSubmit={[Function]}
  >
    <input
      name="name"
      placeholder="Name"
      required={true}
      type="text"
    />
    <input
      name="country"
      placeholder="Country"
      required={true}
      type="text"
    />
    <input
      name="city"
      placeholder="City"
      required={true}
      type="text"
    />
    <textarea
      col="50"
      name="description"
      placeholder="Resort description"
      required={true}
      row="50"
    />
    <input
      name="cost"
      placeholder="Cost"
      required={true}
      type="text"
    />
    <div
      className="file"
    >
      <span
        className="label"
      >
        Enter Resort Image URl:
      </span>
      <input
        className="image_input"
        name="image"
        required={true}
        type="url"
      />
    </div>
    <button
      className="button_resort_add"
      type="submit"
    >
      Update Resort
    </button>
  </form>
</div>
`;
